App icon change

becksen opened the pull request:
update

becksen committed:
Change AppVersion and APP Display Name

becksen committed:
Change HomeRootView Icons for Carbs and Settings

becksen committed:
Add AppIcon Menu under CONFIGURATION in Settings

Jon-b-m committed:
Merge branch 'bdb_dev' into Crowdin

Jon-b-m committed:
Crowdin updates (#599)

Jon-b-m committed:
Missing source string

Jon-b-m committed:
Crowdin updates (#600)

Jon-b-m committed:
Crowdin updates (#601)

Jon-b-m committed:
(Ukrainian) (#602)

Jon-b-m committed:
Crowdin (Russian updates) (#604)

Jon-b-m committed:
Merge branch 'Crowdin' into bdb_dev

becksen committed:
Add AppIcon View translation to locatization
Add SwiftUI views
Enhance RootView and Screen.swift

becksen committed:
Merge branch 'Jon-b-m:bdb_dev' into bdb_dev

becksen committed:
Commit local changes

becksen committed:
commit

becksen committed:
Merge remote-tracking branch 'refs/remotes/origin/bdb_dev'

Conflicts:
	FreeAPS.xcodeproj/project.pbxproj

becksen committed:
Working version switching default and alternateicon

becksen committed:
Merge branch 'bdb' into appIcon_change

becksen committed:
Merge branch 'bdb' into appIcon_change

becksen committed:
Merge remote-tracking branch 'refs/remotes/origin/appIcon_change'

becksen committed:
restore accidently removed icons

Jon-b-m reviewed (changes requested):
Needs some cleaning and refactoring. What does the Test 2.png:s do? Shouldn't it be deleted? Could you please post some screenshots of how the config icon UI Looks like? You have translated the new source string  to "AppIcon Wechseln" for every language. You need to create a new config for Icon change instead of using the autotuneConfig. You changed the HomeRootView Icons for Carbs and Settings.

Jon-b-m commented:
Please remove the references to Autotune and basal profile etc which you copied from
AutotuneConfig, those are not needed. Please revert the changes you made to iPhone deployment version. Please add a screenshot here for the new app icon config. I would like to see how it looks.

becksen committed:
Added icons, Updated SwiftUI View for AppIconChange -> ToDo formatting

Jon-b-m commented:
APP_DISPLAY_NAME = FAX AppIcon
This is name of the app in iOS.

becksen committed:
1st release -> tested with iPhone 11 and iPhone 11 pro

becksen committed:
revert icons in HomeRootView to default

becksen commented:
> Needs some cleaning and refactoring. What does the Test 2.png:s do? Shouldn't it be deleted? Could you please post some screenshots of how the config icon UI Looks like? You have translated the new source string to "AppIcon Wechseln" for every language. You need to create a new config for Icon change instead of using the autotuneConfig. You changed the HomeRootView Icons for Carbs and Settings.

Hi @Jon-b-m, thanks for the remarks. Testimages have been removed. The view has been implemented and reference to autotuneconfig has been removed. HomeRootView icons have been reverted to default.
I've doubts regarding the translation? What is your recommendation to translate the view into the existing languages?

Regards,
becksen.

becksen commented:
> APP_DISPLAY_NAME = FAX AppIcon This is name of the app in iOS.

Displayname has been reverted to FAX default delivery.

becksen commented:
> Please remove the references to Autotune and basal profile etc which you copied from AutotuneConfig, those are not needed. Please revert the changes you made to iPhone deployment version. Please add a screenshot here for the new app icon config. I would like to see how it looks. I see you removed the option to override the default icon with Config or ConfigOverride. Was this necessary?

References have been removed.

Not 100% what you mean by 
"I see you removed the option to override the default icon with Config or ConfigOverride. Was this necessary?"

BR
becksen

becksen reviewed (commented):
This is not required anymore

becksen reviewed (commented):
Not required

Jon-b-m commented:
> Not 100% what you mean by "I see you removed the option to override the default icon with Config or ConfigOverride. Was this necessary?"
> 
> BR becksen

You hard coded the default App Icons to "AppIcon". Before it was a variable you could change in Config and ConfigOverride. Remove this change, please.

becksen committed:
Remove hard coded "AppIcon" on project level. Also checked on targets level. Only thing that is required is to include all available consents.

becksen commented:
> > Not 100% what you mean by "I see you removed the option to override the default icon with Config or ConfigOverride. Was this necessary?"
> > BR becksen
> 
> You hard coded the default App Icons to "AppIcon". Before it was a variable you could change in Config and ConfigOverride. Remove this change, please.

Understood. Reverted accordingly.

Jon-b-m commented:
Still a lot of test icons. I don't understand the need for those?

becksen commented:
> Still a lot of test icons. I don't understand the need for those?
I've removed all testicons. Just a number of different icons have been created as alternative. Being able to change the default purple icon, which is also available in xcassets, the other icons have to be available as well in xcassets. Else you wouldn't have alternative to change to. Hope this is clear now. Added two screenshot to show the dependency of the icons in xcassets. If I get other icons I can add them. But this was selection of 5 different colors. If other colors are desired just let me know.
![Simulator Screen Shot - iPhone 11 Pro - 2023-03-20 at 13 56 25](https://user-images.githubusercontent.com/10520069/226345485-af64576a-a61c-4545-b3af-f460b451f944.png)
<img width="234" alt="Bildschirm­foto 2023-03-20 um 13 57 56" src="https://user-images.githubusercontent.com/10520069/226345682-15bdb859-5c37-44e3-988b-e3bc9d1b63a2.png">

Jon-b-m commented:
The grey test Icons "named Test..." are still there. I have the BW and the purple images already, so you shouldn't have to convert and reduce the resolution? And the other icons are not clean and crisp now. Pixels (noise) in wrong colours and blurry.

Jon-b-m commented:
Could you make the icons bigger? Too small now in the icon setting view

becksen committed:
Removed obsolete purple icons in AppIcon icons

becksen committed:
Clean-Up new iconsents (blue, red, green)
add sizes that make icon appear in better shape when scaled

becksen commented:
> The grey test Icons "named Test..." are still there. I have the BW and the purple images already, so you shouldn't have to convert and reduce the resolution? And the other icons are not clean and crisp now. Pixels (noise) in wrong colours and blurry.

This is a wrong assumption. Purple and Black is already there, but I haven't touched them. This is from original repo. I assume it is not sufficiently assigned. Means there is just one icon for iOS the others are all mapped to MacOS. My new icons have been properly sized and assigned. Hence they look better when scaled in SwifUI. I can change it for the default icons Purple and Black/White when accepted. Else they will probably look not that nice compared to the new ones.

![Bildschirm­foto 2023-03-20 um 15 40 16](https://user-images.githubusercontent.com/10520069/226375322-70da2332-55a8-4031-b72d-73bac31c792b.png)
<img width="675" alt="Bildschirm­foto 2023-03-20 um 15 39 49" src="https://user-images.githubusercontent.com/10520069/226375343-87214ce7-8105-4526-860d-c285ee8224fe.png">

Jon-b-m commented:
I meant in your screenshot. You still have "Test 2" gray icons. in your PR

Jon-b-m commented:
Are the Watch Icons changed also with this setting?

Jon-b-m commented:
A segmented Picker with Icon images perhaps would look better? And we don't need the Xcode names of icons or the orange borders. Could you please polish the UI a bit?

becksen committed:
Remove obsolete .pngs from xcassts
increase icon size

becksen committed:
delete icons

becksen commented:
> I meant in your screenshot. You still have "Test 2" gray icons. in your PR

I've cleaned it up. All Icons contain Test or Test2 have been deleted. I don't see it anymore in the fork.

becksen commented:
> Are the Watch Icons changed also with this setting?

No didn't find any way to do it for watchOS. Only the app icon on iPhone is changed, affected

Jon-b-m commented:
> > Are the Watch Icons changed also with this setting?
> 
> No didn't find any way to do it for watchOS. Only the app icon on iPhone is changed, affected

You change the Watch Icons in Xcode here: 
<img width="1408" alt="image" src="https://user-images.githubusercontent.com/53905247/226390147-d7197e09-b3b5-468d-a763-812b184f1ef6.png">

Jon-b-m commented:
> > I meant in your screenshot. You still have "Test 2" gray icons. in your PR
> 
> I've cleaned it up. All Icons contain Test or Test2 have been deleted. I don't see it anymore in the fork.

<img width="1420" alt="image" src="https://user-images.githubusercontent.com/53905247/226390514-d60358e0-6b91-4edd-a3cb-14c3dfd2500d.png">

becksen committed:
consider colorscheme for text color

becksen committed:
Merge branch 'Jon-b-m:main' into appIcon_change
